lib/services/elasticsearch.js

```diff
@@ -1153,7 +1153,20 @@ class ElasticSearch extends Service {
       }
     }
 
-    return this._mexecute(esRequest, toImport, extracted.rejected);
+    return this._mexecute(esRequest, toImport, extracted.rejected)
+      .then(response => {
+        // with _source: true, ES returns the updated document in
+        // response.result.get._source
+        // => we replace response.result._source with it so that the notifier
+        // module can seamlessly process all kind of m* response*
+        for (const result of response.result) {
```

Comment:
I think @thomasarbona was talking about doing this:
```suggestion
        let j; // NOSONAR
        const res = response.result;
        for (j = 0; j < res.length; j++) {
          if (res[j].get && res[j].get._source) {
            res[j]._source = res[j].get._source;
```
You sacrificed Node.js 6 performance for the sake of readability when you remove this:
```js
        let j; // NOSONAR
        for (j = 0; j < response.result.length; j++) {
          if (response.result[j].get && response.result[j].get._source) {
            response.result[j]._source = response.result[j].get._source;
```

Reply:
last time I checked, `for..of` runs over 2.5M ops/s which should be largely enough here

Reply:
ok I was a bit dishonest, m* routes will probably have to handle millions of documents so we should use the fastest loops available...